### DISCUSSION TITLE
Optional Fit Parameters

### DISCUSSION THREAD

**Discussion by christianaz** [FrandsenGroup/beams#103] [Ideas]
So something that would be really helpful in the fitting part of BEAMS is the option for some parameters to be fixed for some runs, but be able to run free for other runs.

Currently, if I check the "Fixed-Run dependent" box at one fit, it is automatically checked for all the fits I am doing. (If I uncheck it at one temperature, it is unchecked for everything else as well) This is frustrating if I have variables I want fixed at some temperatures but I want to run free at other temperatures. 

It would probably make sense that the box is checked or unchecked depending on which run you select. You would have to check it for all runs that you want fixed. 

It also might be a good idea to have the same feature available for the "Global Fits" checkbox (?)

Let me know if this makes sense :)

**Comment by aPeter1** [comment]
@benfrandsen Changing the checkbox for fixed run dependent would make sense to me. For marking a parameter as global it doesn't quite make sense in my mind to make that decision separately for each run.

**Comment by benfrandsen** [comment]
@aPeter1 So now I'm having this vision of being able to match each parameter in the fitting function with arbitrary sets of runs in the batch to be refined as a common parameter for that set. This would provide maximum flexibility for how the refinements are done. However, this would probably take a fair amount of effort to set up, so we can keep that in mind as a future enhancement. For now, I think it is fine to designate each parameter in the fitting function either as global (i.e. common for all runs in the batch) or as run-specific. Whether global or run-specific, the user should have the option to include the parameter as a variable in the fit or to fix the parameter to a specific value (and for run-specific fixed parameters, to allow different fixed values for each run). I think we could accomplish this without too many changes to the current structure. Sorry for not making this clear enough from the start!

**Reply by aPeter1** [reply]
Coolio, so as I understand this sentence:

"Whether global or run-specific, the user should have the option to include the parameter as a variable in the fit or to fix the parameter to a specific value (and for run-specific fixed parameters, to allow different fixed values for each run)"

this should be how we currently have it. If you check "Fixed-Run Dep" the value you put in the "Value" column next to it will be only applied to the run have selected (selected meaning clicked in the list below, it doesn't have to be checked). 

As far as the other enhancement goes that shouldn't be difficult with the way we have it set up on the backend, the hard part would be figuring out how we want the user to be able to do that on the front end.

**Reply by benfrandsen** [reply]
@christianaz can clarify, but from what she has done in BEAMS, it seems that we do not currently have the possibility of having a run-specific parameter be fixed for some runs and free for others. I'll double check it.

**Reply by aPeter1** [reply]
Sorry, misunderstood what you said, they can't be free. #FreeTheParameters2021 I'll work on that!

**Reply by benfrandsen** [reply]
Thanks for the next viral hashtag :)
